lasagne/theano_extensions/conv.py

```diff
@@ -37,7 +37,11 @@ def conv1d_sc(input, filters, image_shape=None, filter_shape=None,
     conved = T.nnet.conv2d(input_sc, filters_sc, image_shape=image_shape_sc,
                            filter_shape=filter_shape_sc,
                            subsample=(1, subsample[0]))
-    return conved[:, :, 0, :]  # drop the unused dimension
+    try:
+        conved = conved.dimshuffle(0, 1, 3)  # drop the unused dimension
+    except ValueError:
+        conved = T.addbroadcast(conved, 2).dimshuffle(0, 1, 3)
+    return conved
```

Comment:
Yeah, I'm not sure if this is even worth keeping since it wasn't a complete fix... I need to profile and see if it executes any faster (the generated graph does look a bit simpler).